source/grid/tria.cc

```diff
@@ -4512,7 +4512,14 @@ namespace internal
                  triangulation.active_cell_iterators_on_level(level))
               if (cell->refine_flag_set())
                 {
-                  if (cell->at_boundary())
+                  // check if cell is at en EXTERIOR boundary
+                  bool flag = false;
+
+                  for (const auto &f : cell->face_iterators())
+                    flag |=
+                      (f->boundary_id() != numbers::internal_face_boundary_id);
+
+                  if (flag && cell->at_boundary())
```

Comment:
These two conditions are the same, if you look at what `cell->at_boundary()` does.

Reply:
It is not:

https://github.com/dealii/dealii/blob/7a2cec9c87a368a978374001666315c60bc0b003/source/grid/tria_accessor.cc#L2862-L2870

one is checking the boundary ids, the other the index of the neighbors.

Reply:
Ah, I see. That's a nuisance. This seems wrong in the p::f::T case. We should be marking these faces with `internal_face_boundary_id` and make this function return `false`. Is the former what this patch does?